Mono.Debugging/Mono.Debugging.Client/Catchpoint.cs

```diff
@@ -85,6 +85,17 @@ internal override XmlElement ToXml (XmlDocument doc, string baseDir)
 			set { includeSubclasses = value; }
 		}
 
+		/// <summary>
+		/// When an exception first happens, we are given the frame of the exception.
+		/// However by the time the user chooses "Ignore this location" in the exception dialog,
+		/// the current frame might be a different frame (the one with source code),
+		/// and not necessarily the bottom frame of the stack.
+		/// Make sure we remember the original frame where the exception happened
+		/// so that when it happens next time, we use the same frame signature to compare
+		/// against.
+		/// </summary>
+		public string CurrentLocationSignature { get; set; }
```

Comment:
I wonder if we could use a smarter mechanism here, instead of string comparison by assigning IDs (mvid, method token, etc) and associating by that.

Reply:
Yeah, it could just be a struct here, `struct IdentitySignature { FileName, MethodName, Address }`

Reply:
No, because different debuggers get different information. VSCode just sends us a string, with module name already prepended. They also don’t give us the IL offset at all. This information needs to be free form, to enable all debuggers, and leave the string format up to each debugger to interpret at will. It’s also future proof.